### DISCUSSION TITLE
PDF doesn't open in Fastmail

### DISCUSSION THREAD

**Discussion by sebw** [qutebrowser/qutebrowser#6744] [Q&A] [answered]
Hi!

Fastmail user here.

Opening a PDF inside Qutebrowser doesn't seem to work for me.

I did `:set content.pdfjs true` with no improvement. Is there another option or config that I need to enable?

RPM from Fedora 34.

Thanks!

**Comment by The-Compiler** [comment]
Can you elaborate? Where/how exactly should the PDF be opened, and what does "doesn't seem to work" mean exactly? Perhaps a screenshot would help.

**Comment by sebw** [comment]
I should indeed have attached a screenshot, apologies :-) 

So any PDF attached to an email should open up in the area marked in red:

![image](https://user-images.githubusercontent.com/2285094/137597432-0e112b81-6ea1-4408-bf4c-a9caf5b1b436.png)

devtool seems to indicate a 404 when clicking the PDF:

![image](https://user-images.githubusercontent.com/2285094/137597524-570ec30b-fd3e-4489-88c1-e1673cd3073d.png)

![image](https://user-images.githubusercontent.com/2285094/137597604-5ffb2e36-a248-4618-946f-d53eab9c85e4.png)

In Firefox the request is successful:

![image](https://user-images.githubusercontent.com/2285094/137597559-ed4d34e9-1d7c-4ac7-b454-f90f3a3f4bab.png)

Thanks!

**Comment by The-Compiler** [comment, marked as answer]
I don't think the 404 is necessarily related - especially since it's to a different URL (`/jmap/proxy/` and apparently not related to the PDF, rather than `/jmap/download/` for the PDF).

I think this is because PDF.js (or rather how it is integrated in qutebrowser) doesn't support PDFs embedded into pages. I'm guessing if you click "Download" and press Ctrl+p to launch PDF.js, it gets displayed fine? To display embedded PDF's, we should switch to Chromium's own PDF viewer (PDFium), see #4911.
